test/plugin.test.ts

```diff
@@ -0,0 +1,184 @@
+      expect(result.errors).toHaveLength(0);
+    }
+  });
+  test("if one plugin has been added, all the other are exactly the same", () => {
+    if (plugins.length > lastVersion.length) {
+      for (const data of lastVersion) {
+        expect(plugins).toContainEqual(data);
+      }
+    }
+  });
+  test("one plugin is updated if there is an update", () => {
+    if (plugins.length == lastVersion.length) {
+      // get the number of plugin where the version has been updated in one line
+      const updatedPlugins = plugins.filter(
+        (plugin, index) =>
+          plugin.version != lastVersion[index].version &&
+          plugin.name == lastVersion[index].name
+      );
+      const updatedPluginsAmount = updatedPlugins.length;
+      if (updatedPluginsAmount === 0) {
+        throw new Error("No plugin has been updated");
+      } else if (updatedPluginsAmount > 1) {
+        throw new Error("More than one plugin have been updated");
+      }
+      if (updatedPluginsAmount == 1) {
+        // get the plugin that has been updated in the last version
+        const lastVersionUpdatedPlugin = lastVersion.filter(
+          (plugin, index) => plugin.version != plugins[index].version
+        )[0];
+        // the version has to be newer
+        expect(
+          isVersionBiggerThan(
+            updatedPlugins[0]["version"],
+            lastVersionUpdatedPlugin["version"]
+          )
+        ).toBe(true);
+      }
+
+      //  all the other plugins remains the same except the updated one
+      const notUpdatedPlugins = plugins.filter(
+        (plugin, index) =>
+          plugin.version == lastVersion[index].version &&
+          plugin.name == lastVersion[index].name
+      );
+      for (const data of notUpdatedPlugins) {
+        expect(plugins).toContainEqual(data);
+      }
+    }
+  });
+  test("duplicated plugins are not allowed", () => {
+    // filter unique elements in plugins
+    const uniquePlugins = plugins.filter(
+      (plugin, index) => plugins.indexOf(plugin) == index
+    );
+    expect(uniquePlugins).toHaveLength(plugins.length);
+  });
+});
+
+describe("tests on updated or new plugin", () => {
+  // define the list of platforms
+  const platforms = [
+    { name: "windows", pattern: /_windows-amd64.exe$/ },
+    { name: "linux", pattern: /_linux-amd64$/ },
+    { name: "macos-arm64", pattern: /_darwin-arm64$/ },
+    { name: "macos-amd64", pattern: /_darwin-amd64$/ },
+  ];
+
+  let plugins: any[];
+  let lastVersion: any[];
+  let plugin: any;
+  type PluginAsset = {
+    url: string;
+    checksum: string;
+    files?: string[];
+  };
+
+  let pluginAssets: { [key: string]: PluginAsset } = {};
+
+  beforeAll(async () => {
+    plugins = await jsonfile.readFile("plugins.json");
+
+    // Fetch the last version of the plugins file from GitHub
+    lastVersion = await fetch(
+      "https://raw.githubusercontent.com/massalabs/thyra-plugin-store/main/plugins.json"
+    ).then((res) => res.json());
```

Comment:
Don't you think we should have only one beforeAll by file tests!? Like that we all know that the setup for the down tests is done there. Is it possible split these tests in two files and place the beforeAll only once by file!?

WDYT!?

Reply:
It would allow you to send split MR with less code for review and tests segmented. WDYT!?

Reply:
yes It would give more visibility